src/domain/orders/details/rma-sub-modals/products.tsx

```diff
@@ -0,0 +1,316 @@
+import { useAdminVariants } from "medusa-react"
+import React, { useContext, useEffect, useMemo, useState } from "react"
+import Button from "../../../../components/fundamentals/button"
+import Modal from "../../../../components/molecules/modal"
+import { LayeredModalContext } from "../../../../components/molecules/modal/layered-modal"
+import Table, { TablePagination } from "../../../../components/molecules/table"
+import { useTable, usePagination, useRowSelect } from "react-table"
+import StatusIndicator from "../../../../components/fundamentals/status-indicator"
+import { useDebounce } from "../../../../hooks/use-debounce"
+import CheckIcon from "../../../../components/fundamentals/icons/check-icon"
+import clsx from "clsx"
+
+const getProductStatusVariant = (status) => {
+  switch (status) {
+    case "proposed":
+      return "warning"
+    case "published":
+      return "success"
+    case "rejected":
+      return "danger"
+    case "draft":
+    default:
+      return "default"
+  }
+}
+
+type RMASelectProductSubModalProps = {
+  onSubmit: (selectItems) => void
+  selectedItems?: any
+  isLargeModal?: boolean
+}
+
+const IndeterminateCheckbox = React.forwardRef(
+  ({ indeterminate, className, checked, ...rest }, ref) => {
+    const defaultRef = React.useRef()
+    const resolvedRef = ref || defaultRef
+
+    React.useEffect(() => {
+      resolvedRef.current.indeterminate = indeterminate
+    }, [resolvedRef, indeterminate])
+
+    return (
+      <div className="items-center h-full flex">
+        <div
+          onClick={() => resolvedRef.current?.click()}
+          className={`w-5 h-5 flex justify-center text-grey-0 border-grey-30 border cursor-pointer rounded-base ${
+            checked && "bg-violet-60"
+          }`}
+        >
+          <span className="self-center">
+            {checked && <CheckIcon size={16} />}
+          </span>
+        </div>
+        <input
+          type="checkbox"
+          className={clsx("hidden", className)}
+          ref={resolvedRef}
+          {...rest}
+        />
+      </div>
+    )
+  }
+)
+
+const RMASelectProductSubModal: React.FC<RMASelectProductSubModalProps> = ({
+  onSubmit,
+  selectedItems,
+  isLargeModal = true,
+}) => {
+  const { pop } = useContext(LayeredModalContext)
+  const [query, setQuery] = useState("")
+  const [limit, setLimit] = useState(5)
```

Comment:
why 5? And if I change this to 1 then it does not allow me to go to another page, so I am not sure that pagination is working here

Reply:
setLimit is also never used

Reply:
pagination doesn't work because list variants doesn't return the correct count, I have added it to the running issue list of stuff to fix